src/main/java/domain/Cars.java

```diff
@@ -0,0 +1,61 @@
+package domain;
+
+import utils.RandomNumberGenerator;
+import vo.Name;
+import vo.Position;
+import dto.Result;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class Cars {
+    private final List<Car> cars;
+    private final RandomNumberGenerator randomNumberGenerator;
+
+    public Cars(List<Car> cars, RandomNumberGenerator randomNumberGenerator) {
+        this.cars = cars;
+        this.randomNumberGenerator = randomNumberGenerator;
+    }
+
+    public void saveCar(Car car) {
+        cars.add(car);
+    }
+
+    public Position getPosition(Car car) {
+        return car.getPosition();
+    }
+
+    public void move() {
+        cars.forEach((car) -> car.move(randomNumberGenerator.generateRandomNumber()));
+    }
+
+    public List<Name> getWinnerNames() {
+        Position highestPosition = calculateHighestPosition();
+        return cars.stream()
+                .filter(car -> hasHighestPosition(highestPosition, car))
+                .map(Car::getName).collect(Collectors.toUnmodifiableList());
```

Comment:
```suggestion
                .map(Car::getName).
                collect(Collectors.toUnmodifiableList());
```
스트림은 한줄의 점 하나로 사용하면 더 보기 편합니다 :)